app/src/components/billing/BillingContentTabs/PaymentMethods/AddPaymentMethodButton.tsx

```diff
@@ -0,0 +1,221 @@
+import { useState } from "react";
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  HStack,
+  VStack,
+  Icon,
+  Text,
+  Button,
+  Box,
+  useDisclosure,
+  type UseDisclosureReturn,
+  Spinner,
+} from "@chakra-ui/react";
+
+import { useHandledAsyncCallback, useSelectedProject } from "~/utils/hooks";
+import { api } from "~/utils/api";
+import { BsPlus } from "react-icons/bs";
+import { Elements, PaymentElement, useElements, useStripe } from "@stripe/react-stripe-js";
+import { loadStripe } from "@stripe/stripe-js";
+import { toast } from "~/theme/ChakraThemeProvider";
+import { maybeReportError } from "~/utils/errorHandling/maybeReportError";
+import { env } from "~/env.mjs";
+
+export default function AddPaymentMethodButton() {
+  const disclosure = useDisclosure();
+  const selectedProject = useSelectedProject().data;
+
+  const createStripeCustomerMutation = api.payments.createStripeCustomer.useMutation();
+  const createStripeIntentMutation = api.payments.createStripeIntent.useMutation();
+
+  const [clientSecret, setClientSecret] = useState("");
+
+  const [addPaymentMethod, addPaymentMethodLoading] = useHandledAsyncCallback(async () => {
+    if (!selectedProject) return;
+
+    const res = await createStripeCustomerMutation.mutateAsync({
+      projectId: selectedProject.id,
+    });
+
+    if (maybeReportError(res)) return;
+
+    const { clientSecret } = await createStripeIntentMutation.mutateAsync({
+      projectId: selectedProject.id,
+    });
+
+    if (clientSecret) {
+      setClientSecret(clientSecret);
+      disclosure.onOpen();
+    }
+  }, [
+    selectedProject,
+    createStripeCustomerMutation,
+    createStripeIntentMutation,
+    disclosure.onClose,
+    setClientSecret,
+  ]);
+
+  return (
+    <>
+      <Button colorScheme="blue" onClick={() => addPaymentMethod()} w="240px">
+        <HStack spacing={0}>
+          {addPaymentMethodLoading ? (
+            <Spinner />
+          ) : (
+            <>
+              <Icon as={BsPlus} boxSize={6} strokeWidth={0.8} />
+              <Text>Add Payment Method</Text>
+            </>
+          )}
+        </HStack>
+      </Button>
+      <PaymentDetailsModal clientSecret={clientSecret} disclosure={disclosure} />
+    </>
+  );
+}
+
+// Call `loadStripe` outside of a component’s render to avoid recreating the `Stripe` object on every render.
+const stripePromise = loadStripe(
+  "pk_test_51OcEiyIpUtvR6wbgpoMtjp7GzrWoNcjM2kLSeYlEAcP9BevVtv69TeUvhndrg87A4zigWNXYfTjeHyDqX4dt3Pm100pZ8BTBtu",
```

Comment:
Should this go in our environment variables?

Reply:
Oh, I was testing and forgot to replace it!